build_tools/benchmarks/export_benchmark_config.py

```diff
@@ -23,28 +23,75 @@
 # Add build_tools python dir to the search path.
 sys.path.insert(0, str(pathlib.Path(__file__).parent.with_name("python")))
 
+from typing import Callable, Dict, List
 import argparse
 import collections
 import dataclasses
 import json
 
 from benchmark_suites.iree import benchmark_collections
+from e2e_test_framework.definitions import common_definitions, iree_definitions
 from e2e_test_framework import serialization
 from e2e_test_artifacts import iree_artifacts
 
 
+def get_device_spec(
+    config: iree_definitions.E2EModelRunConfig
+) -> common_definitions.DeviceSpec:
+  return config.target_device_spec
+
+
+PresetMatcher = Callable[[iree_definitions.E2EModelRunConfig], bool]
+BENCHMARK_PRESET_MATCHERS: Dict[str, PresetMatcher] = {
+    "x86_64":
+        lambda config: get_device_spec(config).architecture.architecture ==
+        "x86_64",
+    "cuda":
+        lambda config: get_device_spec(config).architecture.architecture ==
+        "cuda",
+    "android-cpu":
+        lambda config:
+        (get_device_spec(config).architecture.type == common_definitions.
+         ArchitectureType.CPU and get_device_spec(
+             config).host_environment.platform == "android"),
+    "android-gpu":
+        lambda config:
+        (get_device_spec(config).architecture.type == common_definitions.
+         ArchitectureType.GPU and get_device_spec(
+             config).host_environment.platform == "android"),
```

Comment:
Why do you need a function just to access a field on the object?

```suggestion
    "x86_64":
        lambda config: config.target_device_spec.architecture.architecture ==
        "x86_64",
    "cuda":
        lambda config: config.target_device_spec.architecture.architecture ==
        "cuda",
    "android-cpu":
        lambda config:
        (config.target_device_spec.architecture.type == common_definitions.
         ArchitectureType.CPU and config.target_device_spec.host_environment.platform == "android"),
    "android-gpu":
        lambda config:
        (config.target_device_spec.architecture.type == common_definitions.
         ArchitectureType.GPU and config.target_device_spec.host_environment.platform == "android"),
```

Reply:
I was thinking it makes lines shorter and easier to read, but actually not. Removed.